filters/include/pcl/filters/impl/filter_indices.hpp

```diff
@@ -100,9 +100,9 @@ pcl::FilterIndices<PointT>::applyFilter (PointCloud &output)
   }
   else
   {
-    output.is_dense = true;
     applyFilter (indices);
     pcl::copyPointCloud (*input_, indices, output);
+    output.is_dense = true;
```

Comment:
Why is it necessary to put this last?

Reply:
When we copy from the original pointcloud, it also copy is_dense from the original cloud, which could be false, but after this filtering it should be true.
But I'm not sure if all filters should remove nan values if doesn't have to keep the output cloud organized.

Reply:
I checked and not all subclasses of `FilterIndices` guarantee that `indices` contain no invalid/nan points. E.g. `RandomSample` may sample invalid points. Also for some other classes this might not be guaranteed if `negative_==true`. Maybe it is best if we leave the code in this file unchanged (or delete `output.is_dense = true;`), and in the future we can consider modifying `copyPointCloud` to check if the output cloud is now dense.

> But I'm not sure if all filters should remove nan values if doesn't have to keep the output cloud organized.

I think it makes sense to remove nan values to stay consistent with `keep_organized_==false`. Also, `user_filter_value_` may be anything (not necessarily nan).